### PR TITLE
added dto,mapper,book controller

### PR DESCRIPTION
**Pull request by DmytroHadiuchko**

### CODE REVIEW

File: src/main/java/dmytro/hadiuchko/springboot/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/books")
```
Comment (oksana-miazina) [inline src/main/java/dmytro/hadiuchko/springboot/controller/BookController.java:16]: The requirements specify that endpoint url should be `/api/books`

**Review by oksana-miazina — APPROVED** [review]
Great!

File: src/main/java/dmytro/hadiuchko/springboot/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -10,15 +14,26 @@
...
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id: " + id));
+        return bookMapper.toDto(book);
```
Comment (karma-o) [inline src/main/java/dmytro/hadiuchko/springboot/service/impl/BookServiceImpl.java]: just a liitle suggestion for you to use
```suggestion
        return bookRepository.findById(id)
                .map(bookMapper::toDto)
                .orElseThrow(
                    () -> new EntityNotFoundException("Can't find book by id: " + id));
```

**Review by karma-o — APPROVED** [review]
Well done, left a small suggestion 😉

**Pull request merged**